be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp

```diff
@@ -343,6 +343,13 @@ Status VerticalSegmentWriter::_append_block_with_partial_content(RowsInBlock& da
     {
         std::shared_lock rlock(tablet->get_header_lock());
         specified_rowsets = tablet->get_rowset_by_ids(&_mow_context->rowset_ids);
+        if (specified_rowsets.size() != _mow_context->rowset_ids.size()) {
```

Comment:
The method `_append_block_with_partial_content` of VerticalSegmentWriter should be removed, it's useless for this class, you can submit another pull request to do such cleanup